zulipterminal/ui_tools/views.py

```diff
@@ -942,27 +941,14 @@ def __init__(self, controller: Any, msg: Message) -> None:
                          for reaction in msg['reactions']])
             reactions[-1] = reactions[-1].rstrip("\n")
 
-        msg_info = OrderedDict([
-            ('Date & Time', time.ctime(msg['timestamp'])[:-5]),
-            ('Sender', msg['sender_full_name']),
-            ('Sender\'s Email ID', msg['sender_email']),
-            ('Reactions', reactions if msg['reactions'] else '---None---'),
-            ])
-
-        widths = [(len(field) + 7,
-                  max(len(reaction_users) for reaction_users in data)
-                  if isinstance(data, list)
-                  else len(data) + 2)
-                  for field, data in msg_info.items()]
-        max_widths = [max(width) for width in zip(*widths)]
-        width = sum(max_widths)
-
-        msg_info_content = [urwid.AttrWrap(
-                urwid.Columns([
-                    urwid.Text(field),
-                    (max_widths[1], urwid.Text(data))
-                ], dividechars=2),
-                None if index % 2 else 'popup_contrast')
-             for index, (field, data) in enumerate(msg_info.items())]
-
-        super().__init__(controller, msg_info_content, 'MSG_INFO', width)
+        msg_info = [
+            ('', [('Date & Time', time.ctime(msg['timestamp'])[:-5]),
+                  ('Sender', msg['sender_full_name']),
+                  ('Sender\'s Email ID', msg['sender_email']),
+                  ('Reactions', reactions if msg['reactions']
+                   else '---None---')]),
```

Comment:
While I know that this code is replaced when you use a category instead, I'm a little concerned that the layout code for columns doesn't work in general, in case we end up with a wide column again? Or is something else happening here?

Reply:
In this particular case, `reactions`, the data structure, that is being passed isn't what `make_table_with_categories` expects. 

However, since the old code uses a hardcoded width (`column_widths[1]`), this is likely to break for wide columns.

A fix could be to fall back `('weight', 1, urwid.Text(data))` whenever the maximum allowed width is smaller than `column_widths[1]`. Though, it would require `max_cols` to be passed into the method.

I couldn't come up with a way to pass maximum allowed width into `calculate_table_witdhs` without tampering the 'flow' of the commits in this PR.

I was thinking of passing a `maximum_width` argument, where `maximum_width=controller.maximum_popup_dimensions()[0]` and then making further amendments to `calculate_popup_width` and `make_table_with_categories` accordingly.

Reply:
This seems to be mostly fixed by the updated commit in any case :+1: